File: Python/rotateArray.py
#A solution that uses a pivot point to rotate the list
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        while k > len(nums):
            k -= len(nums)
        pivot = len(nums) - k
        nums.extend(nums[:pivot])
        del nums[:pivot]

#A solution that transfer the list to a new list then back
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        rot_list = []
        while len(nums) < k:
            k = k - len(nums)
        for x in range(len(nums)):
            rot_list.append(nums[x - k])
        nums[:] = rot_list

File: Kotlin/Array/String/rotateArray.kt
class Solution {
    fun rotate(nums: IntArray, k: Int): Unit {
        if (nums.size == 1 || k == 0){
            return
        }
        var pivot = k
        while (pivot >= nums.size){
            pivot = pivot - nums.size
        }
        if (pivot == 0) {return}
        pivot = nums.size - pivot
        var tempArray = nums.slice(0..pivot-1)
        for (i in (0..nums.size-pivot-1)){
            nums[i] = nums[pivot+i]
        }
        for (i in (0..tempArray.size-1)){
            nums[i + nums.size - pivot] = tempArray[i]
        }
    }
}